### PR TITLE
RCM-103: Added the death date to the screen and implementation methods while maintaining the old API methods too

### PR DESCRIPTION
**Pull request by ssmusoke**

### CODE REVIEW

**Comment by ssmusoke** [comment]
@dkayiwa here is the cohort builder issue - when I add Death Date values and submit a search it returns all patients. This works on Platform 2.x

**Comment by dkayiwa** [comment]
@ssmusoke this is the method that you need to modify by adding filters for date of death: https://github.com/openmrs/openmrs-module-reportingcompatibility/blob/master/api/src/main/java/org/openmrs/report/db/hibernate/HibernateReportDAO.java#L622-L699

**Comment by ssmusoke** [comment]
@dkayiwa is this in addition to the other changes that I made? Or do I have
to revert them too?

**Comment by dkayiwa** [comment]
The other ones are not needed. Unless you of course confirm that they are. 😊

**Comment by ssmusoke** [comment]
@dkayiwa Actually that is already done here https://github.com/ssmusoke/openmrs-module-reportingcompatibility/blob/3fcc24b252ed2659b12c9f5b0642275b607cce02/api/src/main/java/org/openmrs/module/reportingcompatibility/service/db/HibernateReportingCompatibilityDAO.java#L657-L662 the problem seems to be that the parameters are not passed from the UI to the back end

**Comment by dkayiwa** [comment]
@ssmusoke i still stand by my word of the method where you need to add your changes. 😊

**Comment by ssmusoke** [comment]
@dkayiwa the new error I get is https://gist.github.com/ssmusoke/49767360ac9281e0571521a0ce8d5edd

**Comment by dkayiwa** [comment]
How do you reproduce the error? As in what steps?

**Comment by ssmusoke** [comment]
I try to make a search entering two dates of death

**Comment by dkayiwa** [comment]
Is it as easy as using the changes in this pull request to reproduce it?

**Comment by dkayiwa** [comment]
You need to add those two fields in this class: https://github.com/openmrs/openmrs-module-reportingcompatibility/blob/master/api/src/main/java/org/openmrs/reporting/PatientCharacteristicFilter.java#L29-L31

**Comment by ssmusoke** [comment]
@dkayiwa for the pointers now I see that the cohort message is at least customized to include the death date combinations, however it seems like the dates are not passed along ... The combinations of the death date filters match up well as 
screenshot 2017-07-18 20 51 57


However the numbers are for all patients without applying the filters

**Comment by dkayiwa** [comment]
@ssmusoke did you add changes to the DAO method i told you?

**Comment by ssmusoke** [comment]
@dkayiwa I thought I had but stopped at the message signature, and I think I got distracted. Thanks again

**Comment by ssmusoke** [comment]
@dkayiwa I have added both the messages and messages_en_GB.properties files with the translations but they are not being loaded

**Comment by dkayiwa** [comment]
@ssmusoke is there any reason why you added duplicate message properties files in the api? The module already had them in the omod.

**Comment by ssmusoke** [comment]
@dkayiwa I only searched for the messages.properties in api sub=module not the omod. However that is fixed and was the cause of my issue. I have also cleaned up any changes that are not needed for this functionality for your review

File: omod/src/main/webapp/analysis/cohortBuilder.jsp [hunk 1]
```diff
@@ -622,7 +622,7 @@
 				<form name="patient_attribute" method="post" action="cohortBuilder.form">
 					<input type="hidden" name="method" value="addDynamicFilter"/>
```
Comment (dkayiwa) [inline omod/src/main/webapp/analysis/cohortBuilder.jsp]: Do you happen to know the cause of these many red lines?

Comment (ssmusoke) [inline omod/src/main/webapp/analysis/cohortBuilder.jsp]: From https://stackoverflow.com/questions/30667675/what-does-a-red-text-background-mean-in-github-comparison-view it seems like GitHub had an error parsing the file - which I suspect is due to the variable definitions which are non-standard

**Comment by dkayiwa** [comment]
@ssmusoke the ticket status says you are still doing more work on this.

**Comment by ssmusoke** [comment]
@dkayiwa updated ticket status

**Pull request merged**